source/osh/dataconnector/DataConnector.js

```diff
@@ -27,6 +28,20 @@ class DataConnector {
         this.url = url;
         this.id = "DataConnector-" + randomUUID();
         this.reconnectTimeout = 1000 * 60 * 2; //2 min
+        this.status =  Status.DISCONNECTED;
+        this.reconnectionInterval = -1;
+    }
+
+    checkAndclearReconnection() {
+        if(this.reconnectionInterval !== -1) {
+            clearInterval(this.reconnectionInterval);
+            this.reconnectionInterval = -1;
+        }
+    }
+
+    disconnect() {
+        this.checkStatus(Status.DISCONNECTED);
+        this.checkAndclearReconnection();
```

Comment:
When we call `checkAndClearReconnection()` here, it seems that we erase the reconnection interval. Wouldn't that cause the datasource to lose its reconnection capability if we reconnect it later on using connect()?

Reply:
not in that case. Calling this method means that has been called manually. It is a volontary action of the client to disconnect manually the dataSource. In that case, we don't want to reconnect

Reply:
Yes, but even when `disconnect()` is called manually, it doesn't mean we don't want to reconnect it manually later. So it seems to me that, in case we reconnect it manually later by calling `connect()`, we need to make sure the automatic reconnection settings are still the same as what was configured initially.

Reply:
If the connect() is called, it will re-create the reconnection logic when it will enter into onerror or onclose message of the websocket object since the this.reconnectionInterval would be equals to -1. So it should work.